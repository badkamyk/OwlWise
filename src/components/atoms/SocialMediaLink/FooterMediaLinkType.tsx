export type SocialMediaLinkType = {
	href: string;
	srText: string;
	path: JSX.Element;
};
