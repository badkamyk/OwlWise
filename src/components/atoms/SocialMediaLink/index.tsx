import { SocialMediaLinkType } from "./SocialMediaLinkType";

export default function SocialMediaLink({
	href,
	srText,
	path,
}: SocialMediaLinkType) {
	return (
		<a href={href} className="text-gray-200 hover:text-gray-500">
			<span className="sr-only">{srText}</span>
			<svg
				className="h-6 w-6"
				aria-hidden="true"
				fill="currentColor"
				viewBox="0 0 24 24"
			>
				{path}
			</svg>
		</a>
	);
}
