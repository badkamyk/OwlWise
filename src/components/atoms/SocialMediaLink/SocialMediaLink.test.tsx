import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import SocialMediaLink from "./index";

it("SocialMediaLink with props", () => {
	const { container } = render(
		<SocialMediaLink
			href="facebook.com"
			srText={"Facebook link"}
			path={<path d="M0 0h24v24H0z" fill="none" />}
		/>
	);
	expect(
		container.getElementsByClassName("text-gray-200 hover:text-gray-500")
	).toBeTruthy();
	expect(container.getElementsByClassName("sr-only")).toBeTruthy();
	expect(screen.getByText("Facebook link")).toBeTruthy();
	expect(screen.getByRole("link")).toHaveProperty("href", "facebook.com");
	expect(container).toMatchSnapshot();
});
