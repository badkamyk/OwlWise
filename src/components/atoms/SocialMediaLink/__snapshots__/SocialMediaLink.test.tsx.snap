// Vitest Snapshot v1

exports[`SocialMediaLink with props 1`] = `
<div>
  <a
    class="text-gray-200 hover:text-gray-500"
    href="facebook.com"
  >
    <span
      class="sr-only"
    >
      Facebook link
    </span>
    <svg
      aria-hidden="true"
      class="h-6 w-6"
      fill="currentColor"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </a>
</div>
`;
